app/src/main/java/com/github/se/cyrcle/ui/theme/molecules/TopAppBar.kt

```diff
@@ -49,10 +61,57 @@ fun TopAppBar(navigationActions: NavigationActions, title: String, testTag: Stri
               scrolledContainerColor = MaterialTheme.colorScheme.primaryContainer),
       modifier = Modifier.testTag(testTag),
       navigationIcon = {
-        IconButton(
-            onClick = { navigationActions.goBack() },
-            modifier = Modifier.testTag("${testTag}GoBackButton")) {
-              Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Go Back")
-            }
+        Box(modifier = Modifier.padding(8.dp)) {
+          IconButton(
+              onClick = { navigationActions.goBack() },
+              modifier =
+                  Modifier.testTag("${testTag}GoBackButton")
+                      .background(Color.White.copy(alpha = 0.1f), shape = RoundedCornerShape(12.dp))
+                      .size(40.dp)) {
+                Icon(
+                    Icons.AutoMirrored.Filled.ArrowBack,
+                    contentDescription = "Go Back",
+                    modifier = Modifier.size(24.dp))
```

Comment:
I don't understand why here the `IconButton` contains an `Icon` since the `IconButton` can do everything the `Icon` does. Is it a question of size management ?

Reply:
This is just how material3 IconButton composables are. I don't make the rules

Reply:
Icon button has a modifier and a vector Image which makes the Icon useless. I think the Icon here is not necessary unless it is for size management.

Reply:
I think you're talking about the IconButtom from the atoms. It doesn't work for me. I need to use material3's IconButtom

Reply:
Yes, It was the one I was talking about but if it doesn't work then you made the right choice I won't block your PR for this.

Reply:
Btw just to clarify when I say it doesn't work i meant that it doesn't suit my needs. Not that it actually doesn't work